hazelcast/src/main/java/com/hazelcast/jet/impl/connector/StreamEventJournalP.java

```diff
@@ -386,24 +395,41 @@ public int preferredLocalParallelism() {
 
         @Override
         public void init(@Nonnull Context context) {
-            if (clientXml != null) {
-                initRemote();
+            // The order is important.
+            // If dataLinkConfig is specified prefer it to clientXml
+            if (dataLinkRef != null) {
+                HzClientDataLinkFactory hzClientDataLinkFactory = getDataStoreFactory(context.hazelcastInstance(),
+                        dataLinkRef.getName());
+                HazelcastInstance client = hzClientDataLinkFactory.getDataLink();
```

Comment:
If this is a non-shared client, how is it closed?

Reply:
The client created by the dataLinkRef is the shared client. 
It accesses the remote cluster to find the number of partitions. It is not closed